goland/tcp/mod.go

```diff
@@ -83,3 +84,50 @@ func (hs *Service) Execute(snap store.Snapshot, step execution.Step) (execution.
 		Accepted: true,
 	}, nil
 }
+
+// ExecuteGraalvmScalarMultiply execute a scalar multiplication on stored data
+func (hs *Service) ExecuteGraalvmScalarMultiply(snap store.Snapshot, step execution.Step) (execution.Result, error) {
+	res := execution.Result{}
+
+	storedData, err := snap.Get(storeKey[:])
+	if err != nil || len(storedData) == 0 {
```

Comment:
you're doing two different checks that should be separate. One is an error in getting the value, the other is checking that the value exist. For the later one, `snap.Get` will return `nil` if the value does not exist. So if you want that check you should add:

```go
if storedData == nil {
```